docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+// on the Entry must match the selector of the Catalog
+type ResourceCatalogEntry struct {
+	unversioned.TypeMeta
+	api.ObjectMeta
+
+	Spec   ResourceCatalogEntrySpec
+	Status ResourceCatalogEntryStatus
+}
+
+// ResourceCatalogEntrySpec gives information about the Entry and describes how a claim would be satisfied.
+// Only one of the pointers maybe specified.  Only the Description is exposed.
+// An Update can be used to re-detect the exposed resources.  This could ripple through and kick synchronization
+// across all claims.  Otherwise, new claims get new resource and old ones keep their existing ones.
+type ResourceCatalogEntrySpec struct {
+	// Description says what this does
+	Description string
+
+	// LocalResources describes the case where resources exist in this namespace that will be expose via a claim
+	// to another namespace
+	LocalResources *LocalResourceSpec
+
+	// ResourceList describes the case where we have list that we want to create in the claim-namespace, but the
+	// resources don't actually exist in this namespace
+	ResourceList *ResourceListSpec
+
+	// Extension is a spot for a generic spec.  This would need a custom controller to make use of.  Imagine this
+	// as your placeholder for an openshift template.
+	Extension runtime.Object
+}
+
+// LocalResourceSpec describes the case where resources exist in this namespace that will be expose via a claim
+// to another namespace.
+type LocalResourceSpec struct {
+	// Transitive indicates whether "depends-on" annotations in the linked objects are chased and added to Status
+	Transitive bool
+
+	// ExposedResources is the list of resources (name or name/UID) to be included in the exposure
+	ExposedResources []api.LocalObjectReference
+}
+
+// LocalResourceStatus indicates which resources are actually exposed.
+type LocalResourceStatus struct {
+	// ExposedResources is the list of resources complete with name/UID matching to be included in the exposure
+	// This will include all transitive links based on "depends-on" if chosen in Spec.  The UID MUST be specified
+	// so that we don't accidentally expose new resources with the same name as the old resources.
+	ExposedResources []api.LocalObjectReference
+}
+
+// ResourceListSpec describes the case where we have list that we want to create in the claim-namespace, but the
+// resources don't actually exist in this namespace.  You can imagine this being useful for cases where you don't
+// want the resources in your namespace.  RC or DaemonSet maybe.
+type ResourceListSpec struct {
+	// Resource is the list of resources to create to satisfy a claim.
+	Resources api.List
+}
+
+// ResourceListStatus is for congruence with Spec.  This one ought not need anything
+type ResourceListStatus struct {
+	// Resource is the list of resources to create to satisfy a claim.
+	Resources api.List
+}
+
+// ResourceCatalogEntrySpec gives information about the Entry and describes how a claim would be satisfied.
```

Comment:
> I get the generation bit, but what are you using LocalResources and ResourceList for in the status?

Your original proposal liked the idea describing a chain of "depends-on" relationships to link objects.  If you transitively chase those links, you need to track what you actually found.  This shows you the list so that a user can re-kick the detection by bumping spec generation.